nexis/lib/widgets/auth/custom_column.dart

```diff
@@ -0,0 +1,192 @@
+
+  @override
+  Widget build(BuildContext context) {
+    List<double> margins = getMargins(type);
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        heading(type, context, largeLabel),
+        sizedBox(margins[0]),
+        text(type, context, mediumLabel),
+        sizedBox(margins[1]),
+        textField(type, controller, onSubmitted, obscureText, hint, onTap, readOnly),
+        rowCol(type, context, onTap, monthController, dayController, yearController),
+        sizedBox(margins[2]),
+        richText(type, context, mediumBody, recognizer),
+        button(type, onPressed, buttonText),
+        sizedBox(margins[3]),
+        richText2(type, context, smallLabel, mediumBody, recognizer),
+      ],
+    );
+  }
+
+  heading(ColumnType type, BuildContext? context, String? largeLabel) {
+    if (type != ColumnType.type1) { return sizedBox(0); }
+    return Text(
+      largeLabel ?? "", 
+      style: Theme.of(context!).textTheme.labelLarge,
+    );
+  }
+
+  textField(ColumnType type, TextEditingController? controller, void Function(String)? onSubmitted, bool? obscureText, 
+    String? hint, void Function()? onTap, bool? readOnly) {
+    if (type == ColumnType.type4 || type == ColumnType.type5) { return sizedBox(0); }
+    return SizedBox(
+      width: 500,
+      child: CustomTextField(
+        obscureText: obscureText ?? false,
+        hint: hint ?? '',
+        controller: controller,
+        onSubmitted: onSubmitted,
+        onTap: onTap,
+        readOnly: readOnly ?? false,
+      ),
+    );
+  }
+
+  text(ColumnType type, BuildContext? context, String? mediumLabel){
+    if (type == ColumnType.type4) { return sizedBox(0); }
+    return Text(
+      mediumLabel ?? "", 
+      style: Theme.of(context!).textTheme.labelMedium,
+    );
+  }
+
+  sizedBox(double? height) {
+    return SizedBox(height: height);
+  }
+
+  richText(ColumnType type, BuildContext? context, String? mediumBody, GestureRecognizer? recognizer) {
+    if (type != ColumnType.type2) { return sizedBox(0); }
+    return RichText(
+      text: TextSpan(
+        text: mediumBody ?? "", 
+        style: Theme.of(context!).textTheme.bodyMedium,
+        recognizer: recognizer,
+        mouseCursor: SystemMouseCursors.click,
+      ),
+    );
+  }
+  richText2(ColumnType type, BuildContext? context, String? smallLabel, String? mediumBody, GestureRecognizer? recognizer) {
+    if (type != ColumnType.type4) { return sizedBox(0); }
+    return RichText(
+      text: TextSpan(
+        children: [
+          TextSpan(
+            text: smallLabel ?? "", 
+            style: Theme.of(context!).textTheme.labelSmall,
+          ),
+          TextSpan(
+            text: mediumBody ?? "",
+            style: Theme.of(context).textTheme.bodyMedium,
+            recognizer: recognizer,
+            mouseCursor: SystemMouseCursors.click,
+          ),
+        ],
+      ),
+    );
+  }
```

Comment:
Combine these two together, since there's significant code duplication. This could be refactored to use a single method with some conditional logic.